xtask/src/cli.rs

```diff
@@ -0,0 +1,60 @@
+use std::process::exit;
+
+use anyhow::{anyhow, bail, Context};
+use pico_args::Arguments;
+
+const HELP: &str = "\
+Usage: xtask <COMMAND>
+
+Commands:
+  run-wasm
+
+Options:
+  -h, --help  Print help
+";
+
+pub(crate) struct Args {
+    pub(crate) subcommand: Subcommand,
+}
+
+impl Args {
+    pub fn parse() -> Self {
+        let mut args = Arguments::from_env();
+        log::debug!("parsing args: {args:?}");
+        if args.contains("--help") {
+            eprint!("{HELP}");
+            exit(101);
```

Comment:
Remind me what 101 is for?

Reply:
Ah, sorry, this is pure muscle memory that I'm glad you called out. [Cargo uses the 101 exit status to indicate that Cargo has failed](https://doc.rust-lang.org/cargo/commands/cargo.html#exit-status). It's become an (informal, AFAIK) exit status for "the Cargo plugin you called failed, not something underneath that it may have been subprocess'ing to".

The thing is, I don't think this habit makes sense here. It's actually debatable whether a `--help` string should return non-zero (`cargo`, for example, returns 0).

I'm happy to change this to whatever, with whatever docs you want. :)

Reply:
This is interesting, I called a bunch of things and they seem to disagree.

cargo-related things all return 0 on finish.
unix tools (`find` returns 1, `grep` returns 2??) return non-zero

I kinda prefer returning non-zero as returning `-h` in something like a script/CI is almost always should fail.

Reply:
Tldr: 101 is fine

Reply:
I avoided `const` so I didn't have to specify a type, but I extracted a variable + doc link with 32dec0f2.